### PR TITLE
Adding breakpoint g

### PR DESCRIPTION
**Pull request by roeematza**

### CODE REVIEW

File: src/breakpoints/EraseCountAnyBlockBreakpoint.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+			if(blockReachedCount){
+				return true;
+			}
+		}
+		return false; 
+	}
+
+	@Override
+	public String getDisplayName() {
+		return "Any block reaches erease count";
```
Comment (dlvhdr) [inline src/breakpoints/EraseCountAnyBlockBreakpoint.java:33]: typo (erease)

File: src/breakpoints/EraseCountBlockBreakpoint.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+			return false;
+		}
+		
+		return previousDevice.getBlockByIndex(getBlockIndex()).getEraseCounter() != getCount() 
+				&& currentDevice.getBlockByIndex(getBlockIndex()).getEraseCounter() == getCount();			 
+	}
+
+	@Override
+	public String getDisplayName() {
+		return "Block reaches erease count";
```
Comment (dlvhdr) [inline src/breakpoints/EraseCountBlockBreakpoint.java:27]: typo (erease)

File: src/breakpoints/HotColdPartitionHoldsPrecentOfPages.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
+package breakpoints;
+
+import manager.HotColdPartition;
+import entities.Device;
+import entities.hot_cold.HotColdDevice;
+import entities.hot_cold.HotColdPage;
+import general.ConfigProperties;
+
+public class HotColdPartitionHoldsPrecentOfPages extends BreakpointBase {
```
Comment (dlvhdr) [inline src/breakpoints/HotColdPartitionHoldsPrecentOfPages.java:9]: typo - precent -> percent

Comment (dlvhdr) [inline src/breakpoints/HotColdPartitionHoldsPrecentOfPages.java:9]: Shouldn't the breakpoint be: HotCold – partition p holds X% of **blocks**?

File: src/breakpoints/HotColdPartitionHoldsPrecentOfPages.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+	}
+
+	@Override
+	public String getDescription() {
+		return "HotCold partition " + getPartition() + " holds " + getPrecent() + " precent of pages";
+	}
+
+	@Override
+	public void addComponents() {
+		mComponents.add(new BreakpointComponent("precent", int.class, "Precent of pages"));
```
Comment (dlvhdr) [inline src/breakpoints/HotColdPartitionHoldsPrecentOfPages.java:55]: typo (precent -> percent)

File: src/breakpoints/PagesWritenInChip.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+	}
+
+	@Override
+	public String getDisplayName() {
+		return "X pages are writen in chip";
+	}
+
+	@Override
+	public String getDescription() {
+		return getCount() + " pages are writen in chip " + mChipIndex;
```
Comment (dlvhdr) [inline src/breakpoints/PagesWritenInChip.java:31]: typo - writen -> written

File: src/breakpoints/PagesWritenInChip.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+		if(previousDevice == null){
+			return currentDevice.getChip(mChipIndex).getTotalWritten() == mCount; 
+		}
+		return currentDevice.getChip(mChipIndex).getTotalWritten() == mCount 
+				&& previousDevice.getChip(mChipIndex).getTotalWritten() != mCount;
+	}
+
+	@Override
+	public String getDisplayName() {
+		return "X pages are writen in chip";
```
Comment (dlvhdr) [inline src/breakpoints/PagesWritenInChip.java:26]: typo - writen -> written

File: src/breakpoints/PagesWritenInDevice.java [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+	}
+
+	@Override
+	public String getDescription() {
+		return getCount() + " pages are writen in device";
+	}
+
+	@Override
+	public void addComponents() {
+		mComponents.add(new BreakpointComponent("count", int.class, "Number of pages writen"));
```
Comment (dlvhdr) [inline src/breakpoints/PagesWritenInDevice.java:35]: written

File: src/breakpoints/PagesWritenInPlane.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+	}
+
+	@Override
+	public String getDescription() {
+		return getCount() + " pages are writen in plane " + mPlaneIndex;
+	}
+
+	@Override
+	public void addComponents() {
+		mComponents.add(new BreakpointComponent("count", int.class, "Number of pages writen"));
```
Comment (dlvhdr) [inline src/breakpoints/PagesWritenInPlane.java:37]: written

File: src/breakpoints/ReusableLevelPagesPrecent.java [hunk 9]
```diff
@@ -0,0 +1,74 @@
+package breakpoints;
+
+import entities.Device;
+import entities.reusable.ReusableDevice;
+import entities.reusable.ReusablePage;
+import general.ConfigProperties;
+
+public class ReusableLevelPagesPrecent extends BreakpointBase {
```
Comment (dlvhdr) [inline src/breakpoints/ReusableLevelPagesPrecent.java:8]: percent

File: src/entities/Chip.java [hunk 10]
```diff
@@ -131,15 +140,25 @@ protected int getMinValidCountPlaneIndex() {
...
-
-	public Chip<P, B, T> setPlane(int index, T t) {
+		
+		T newPlane = (T) getPlane(index).writeLP(lp, arg);
 		List<T> updatedPlanes = getNewPlanesList();
-		updatedPlanes.set(index, t);
+		updatedPlanes.set(index, newPlane);
 		Builder<P, B, T> builder = getSelfBuilder();
-		builder.setPlanes(updatedPlanes);
+		builder.setPlanes(updatedPlanes).setTotalWritten(getTotalWritten()  + 1);;
```
Comment (elys1um) [inline src/entities/Chip.java]: Does totalWritten ever decrease (for example after cleaning)?

**Comment by dlvhdr** [comment]
Mainly typo mistakes.
Also see comment about totalWritten.

**Pull request closed without merging**